ibc/types/proto/proofs.proto

```diff
@@ -0,0 +1,222 @@
+syntax = "proto3";
+
+// This file is a clone from the github.com/cosmos/ics23 repo, it has been
```

Comment:
1. I suggest even creating a small bash file that does a `wget` to the files you need (e.g. `https://raw.githubusercontent.com/cosmos/ics23/master/proto/cosmos/ics23/v1/proofs.proto`) so we can update it by running one command if it ever changes.

2. Make sure to highlight this comment by starting with a `// !! IMPORTANT !! This file is a clone`

3. What problems are you encountering or questions do you have? From my point of view, embedding the `.proto` definition instead of the compiled `.pb.go` file is the correct way to go about it.

Reply:
So if you look in #916 I need to embed the `ProofSpec` into the `PocketClientState` message. This is only possible when 1. we import the proto file and 2. the type is a message.

As the cosmos gogoproto compiler doesnt produce types that satisy the `proto.Message` interface we cannot just use the `ics23.SmtSpec` type they define we must recompile using normal protoc (so it produces `proto.Message`) types and then create the type ourselves. This way we can use the proto file, import it and then use the type. Adding the conversion allows us to still use the `cosmos/ics23` library for the methods they define but gives us the ability to use the message aswell